### PR TITLE
add support for Lean Theorem Prover

### PR DESCRIPTION
**Pull request by soonhokong**
[Lean code in the wild](https://github.com/search?p=1&q=extension%3Ahlean+extension%3Alean+NOT+nothack&ref=searchresults&type=Code&utf8=%E2%9C%93)

### CODE REVIEW

**Comment by soonhokong** [comment]
FYI, the Travis failure is because of `vendor/grammars/sublime_cobol`:

```
Cloning into 'vendor/grammars/sublime_cobol'...
fatal: unable to access 'https://bitbucket.org/bitlang/sublime_cobol/': The requested URL returned error: 503
Clone of 'https://bitbucket.org/bitlang/sublime_cobol' into submodule path 'vendor/grammars/sublime_cobol' failed
```

https://travis-ci.org/github/linguist/jobs/51851042

**Comment by arfon** [comment]
:+1:

**Comment by arfon** [comment]
Thanks @soonhokong - this change should be live on GitHub in the next couple of weeks.

**Comment by soonhokong** [comment]
@arfon, thank you!

**Comment by soonhokong** [comment]
Dear @arfon,

First of all, thanks again for merging this pull-request and making a new release including this! Our team is very excited to see the changes!

We've found that the ace-mode is not working for `.lean` files yet while `data-ace-mode` is correctly set with `lean`. I guess it is because of the version of ace editor that github is using. FYI, ace-editor team accepted `.lean` and `.hlean` supports about 10 days ago by merging https://github.com/ajaxorg/ace/commit/49ff50463939d1a5577088dfeab80bdb91614602 while their latest official release was at last year November.

In short, could you share with us 1) which version of ace editor github is using, and 2) what is github's policy for that? Ultimately, we want to know when we can see ace editor support for lean files on https://github.com.

Sincerely,
@soonhokong

**Comment by arfon** [comment]
Hi @soonhokong, it looks like there hasn't been a new release of ace [since November](https://github.com/ajaxorg/ace/releases). It might be worth asking the authors to make a new release so we can update our version [here](https://github.com/josh/sprockets-ace).

/ cc @josh incase he has any other suggestions.

**Comment by soonhokong** [comment]
@arfon, thank you!

@josh, could you let us know your policy on updating https://github.com/josh/sprockets-ace repository? Do I need to ask the ace-editor team to make a new release including Lean supports? or could you pull their latest version https://github.com/ajaxorg/ace/commits/master?

**Comment by larsbrinkhoff** [comment]
Hello @soonhokong,

For your information, the ACE grammar for Lean has been removed.  See here: https://github.com/ajaxorg/ace/commit/96e0aba61b8aa4b111e2c90cd905aa0c9aad7c5d, and also #3108.
